src/xrpld/consensus/ConsensusParms.h

```diff
@@ -112,30 +128,68 @@ struct ConsensusParms
     // we increase the threshold for yes votes to add a transaction to our
     // position.
 
-    //! Percentage of nodes on our UNL that must vote yes
-    std::size_t avINIT_CONSENSUS_PCT = 50;
-
-    //! Percentage of previous round duration before we advance
-    std::size_t avMID_CONSENSUS_TIME = 50;
-
-    //! Percentage of nodes that most vote yes after advancing
-    std::size_t avMID_CONSENSUS_PCT = 65;
-
-    //! Percentage of previous round duration before we advance
-    std::size_t avLATE_CONSENSUS_TIME = 85;
-
-    //! Percentage of nodes that most vote yes after advancing
-    std::size_t avLATE_CONSENSUS_PCT = 70;
-
-    //! Percentage of previous round duration before we are stuck
-    std::size_t avSTUCK_CONSENSUS_TIME = 200;
-
-    //! Percentage of nodes that must vote yes after we are stuck
-    std::size_t avSTUCK_CONSENSUS_PCT = 95;
-
     //! Percentage of nodes required to reach agreement on ledger close time
     std::size_t avCT_CONSENSUS_PCT = 75;
+
+    //! Number of rounds before certain actions can happen.
+    // (Moving to the next avalanche level, considering that votes are in a
+    // stable state without consensus.)
+    std::size_t avMIN_ROUNDS = 2;
+
+    //! Number of rounds before a stuck vote is considered unlikely to change
+    //! because voting is in an undesriable stable state
+    std::size_t avSTUCK_VOTE_ROUNDS = 5;
+
+    enum AvalancheState { init, mid, late, stuck };
+    struct AvalancheCutoff
+    {
+        std::size_t const consensusTime;
+        std::size_t const consensusPct;
+        AvalancheState const next;
+    };
+    std::map<AvalancheState, AvalancheCutoff> avalancheCutoffs = {
```

Comment:
To work around std::map copying in the libstdc++ on MacOS:

```suggestion
    const std::map<AvalancheState, AvalancheCutoff> avalancheCutoffs{
```

Reply:
> To work around std::map copying in the libstdc++ on MacOS:

Done